src/lib/components/Pagination.svelte

```diff
@@ -0,0 +1,140 @@
+<script context="module" lang="ts">
+	const PAGE_SIZE = 50
+
+	export function paginate<T extends any[]>(pageParam: string | null, items: T) {
+		const number = pageParam ? parseInt(pageParam) : 1
+		const itemIndexStart = (number - 1) * PAGE_SIZE
+		return {
+			number,
+			items: items.slice(itemIndexStart, itemIndexStart + PAGE_SIZE) as T,
+		}
+	}
+</script>
+
+<script lang="ts">
+	import { goto } from '$app/navigation'
+
+	export let current: number = 1
+	export let results: number
```

Comment:
Personal opinion (so feel free to ignore if you disagree):
I think these should be named `currentPage` and `totalResults` as I prefer verbosity.

Reply:
Agreed!